src/Core/Domain/Customer/Group/QueryResult/EditableCustomerGroup.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core\Domain\Customer\Group\QueryResult;
+
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\ValueObject\GroupId;
+
+class EditableCustomerGroup
+{
+    /**
+     * @var GroupId
+     */
+    private $id;
+
+    /**
+     * @var string[]
+     */
+    private $name;
+
+    /**
+     * @var float
+     */
+    private $reduction;
+
+    /**
+     * @var int
+     */
+    private $priceDisplayMethod;
+
+    /**
+     * @var bool
+     */
+    private $showPrice;
+
+    /**
+     * @param GroupId $id
+     * @param string[] $name
+     * @param float $reduction
+     * @param int $priceDisplayMethod
+     * @param bool $showPrice
+     */
+    public function __construct(
+        GroupId $id,
```

Comment:
```suggestion
        int $id,
```
use scalar values in query results except for DecimalNumber values.
(at least we do it in most of query results) I think the argument to use only scalar values was that query results are for reading only (and rendering in html or so..) and they supposed to be fetched correct from DB, so there is not much point triggering value object constraints or smth like that...

Reply:
Done